library/alloc/src/boxed.rs

```diff
@@ -2080,6 +2082,47 @@ impl<I> FromIterator<I> for Box<[I]> {
     }
 }
 
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<I, const N: usize, A: Allocator> !Iterator for Box<[I; N], A> {}
+
+/// `Box` is fundamental, so coherence needs help understanding these impls are okay.
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<'a, const N: usize, I, A: Allocator> !Iterator for &'a Box<[I; N], A> {}
+
+/// `Box` is fundamental, so coherence needs help understanding these impls are okay.
```

Comment:
This (and the comment above it) are going to show up in public docs right? If so, could these be elaborated on a bit more? Maybe just a short description of what would happen if these impls didn't exist would be sufficient.

Reply:
Ah, yeah I will make these less esoteric.